File: src/lua/nodes/VolumeNode.lua
local BaseNode = require("BaseNode")
local DECLARE_SETTERS_GETTERS = require("settersgetters")

local VolumeNode = {}
VolumeNode._NAME = "VolumeNode"

do
	DECLARE_SETTERS_GETTERS(VolumeNode, {
		{ name = "Gain", type = "number" },
	})
end

return BaseNode.Extend(VolumeNode)

File: src/js/nodes/VolumeNode.js
var BaseNode = require('../BaseNode');

function VolumeNode(ctx, options) {
	BaseNode.call(this, ctx, options);
	this.input = this.output = this.ctx.audio.createGain();
}

VolumeNode.prototype = Object.create(BaseNode.prototype);
VolumeNode.prototype.constructor = VolumeNode;

VolumeNode.prototype.initialize = function(cb) {
	this.input.gain.value = this.options[0];
	if (cb) cb();
}

VolumeNode.prototype.onUpdateOptions = function() {
	this.initialize();
}

module.exports = VolumeNode;